test/test_mp3d_eqa.py

```diff
@@ -227,3 +227,38 @@ def test_mp3d_eqa_sim_correspondence():
             cycles_n -= 1
 
     env.close()
+
+
+def test_eqa_task():
+    eqa_config = get_config(CFG_TEST)
+
+    if not mp3d_dataset.Matterport3dDatasetV1.check_config_paths_exist(
+        eqa_config.DATASET
+    ):
+        pytest.skip("Please download Matterport3D EQA dataset to data folder.")
+
+    dataset = make_dataset(
+        id_dataset=eqa_config.DATASET.TYPE, config=eqa_config.DATASET
+    )
+    env = habitat.Env(config=eqa_config, dataset=dataset)
+    env.episodes = [
+        episode
+        for episode in dataset.episodes
+        if int(episode.episode_id) in TEST_EPISODE_SET[:EPISODES_LIMIT]
+    ]
+
+    for i in range(3):
+        env.reset()
+        env.step(1 + np.random.choice(2))
+        metrics = env.get_metrics()
+        del metrics["episode_info"]
+        print(metrics)
+    correct_answer_id = dataset.get_answers_vocabulary()[
+        env.current_episode.question.answer_text
+    ]
+    env.step(0)
+    env.task.answer_question(correct_answer_id, env.episode_over)
```

Comment:
This won't work well for vectorized environments.

I still think the right structure is to have `env.step(0, task_action=correct_answer_id)`  That'll work nicely for vectorized environments.